util/work_queue_test.cc

```diff
@@ -0,0 +1,254 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
```

Comment:
I only see a unit test for work queue, but I didn't see unit test that builds SST files with parallel compression. Maybe I missed something. Also it is preferred to have at least one unit test that covers the whole flow: from setting the option, generating SST files, and read them back and check the values are correct.

Reply:
Sorry I missed unit tests in former versions.
I leveraged existing RandomizedHarnessTest in `table/table_test.cc` as basic unit tests. They now also check cases where parallel compression is enabled.
I also enabled `DBBasicTestWithParallelIO` in `db/db_basic_test.cc` and `DBTest2::CompressionOptions` in `db/db_test2.cc` to check parallel compression cases.
These tests should all cover the whole flow.